### ISSUE TITLE
No such file or directory (os error 2)

### ISSUE THREAD

**Issue by RensR** [Cyfrin/aderyn#266]
After installing aderyn and running it for the first time on a Foundry project, it output many `No such file or directory (os error 2)` errors. Since this is a fresh install and I supplied no arguments, it doesn't feel like this is a config/user error.

```
➜  contracts git:(golf) ✗ aderyn                     
Detecting framework...
Framework detected: Foundry mode engaged.
Foundry root path: "."

Usage: forge build [OPTIONS]

For more information, try '--help'.
Foundry src path: "./src/v0.8"
Error reading Solidity source file: "src/v0.8/ccip/test/mocks/MockBridgeAdapter.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Error reading Solidity source file: "src/v0.8/automation/v2_1/test/BaseTest.t.sol"
No such file or directory (os error 2)

```

**Comment by TilakMaddy** [comment]
Can you try to exclude the `test` folder by saying `aderyn -e /test/` ? (BTW run `foundryup` before that because `--ast` option was a breaking change introduced by forge

**Comment by RensR** [comment]
Full log 

```
➜  contracts git:(golf) ✗ aderyn --scope v0.8/ccip -e v0.8/ccip/test
Detecting framework...
Framework detected: Foundry mode engaged.
Foundry root path: "."
[⠒] Compiling...
No files changed, compilation skipped
Foundry src path: "./src/v0.8"
Error reading Solidity source file: "src/v0.8/ccip/pools/LockReleaseLiquidityTokenPool.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Error reading Solidity source file: "src/v0.8/ccip/pools/liquidity/ArbitrumL1BridgeAdapter.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Error reading Solidity source file: "src/v0.8/ccip/pools/liquidity/ArbitrumL1BridgeAdapter.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Error reading Solidity source file: "src/v0.8/ccip/libraries/CallWithExactGas.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Error reading Solidity source file: "src/v0.8/ccip/pools/liquidity/interfaces/IBridge.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Error reading Solidity source file: "src/v0.8/ccip/pools/liquidity/interfaces/IBridge.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Error reading Solidity source file: "src/v0.8/ccip/pools/liquidity/ArbitrumL2BridgeAdapter.sol"
No such file or directory (os error 2)

Location:
    /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/lib.rs:149:8
Get Detectors
Running 22 detectors
Detectors run, processing found issues
Found issues processed. Printing report
thread 'main' panicked at /Users/rens/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aderyn_core-0.0.18/src/report/markdown_printer.rs:278:55:
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by RensR** [comment]
Repo it ran from, including the foundry profile 

https://github.com/smartcontractkit/ccip/tree/ccip-develop/contracts

**Comment by TilakMaddy** [comment]
@RensR  I have uploaded the report here https://gist.github.com/TilakMaddy/bf21b4de205195f6606d4e8dd4f4fa23 

Will talk to @alexroan about this  : )

**Comment by alexroan** [comment]
@TilakMaddy what was the action for this one? Was it custom Foundry profile handling? If so, the latest release should fix it.

**Comment by TilakMaddy** [comment]
@alexroan I'm afraid not because if you look closely error here, as shown in the last line is 

> /src/report/markdown_printer.rs:278:55:

Now I know about a bug that exists currently which is that when you choose `stdout` the markdown printer will panic because printer wouldn't be able to calculate relative path to project root when it doesn't know where it has to output the markdown file. (because writer is `stdout`)

However here I don't see any sign that Rens has used stdout. Therefore when I tried it I had not faced an issue

**Comment by RensR** [comment]
Can confirm this is fixed in v0.0.19

**Comment by alexroan** [comment]
Closing this. I'll run on the CCIP repo to see if I get an empty report.
